src/scripts.js

```diff
@@ -70,7 +70,13 @@ function isScriptHashInput(script, allowIncomplete) {
   if (!Buffer.isBuffer(lastChunk)) return false
 
   var scriptSig = Script.fromChunks(script.chunks.slice(0, -1))
-  var scriptPubKey = Script.fromBuffer(lastChunk)
+  var scriptPubKey
+
+  try {
+    scriptPubKey = Script.fromBuffer(lastChunk)
+  } catch (e) {
```

Comment:
I wonder if we should be checking specifically for a `RangeError` vs this catch all...